mice-and-cats-java/src/main/java/foop/world/World.java

```diff
@@ -280,24 +234,34 @@ public void render(Graphics2D g, int w, int h, String playerName, boolean superV
             }
         }
 
+        // Draw entities (Cat's and Mice)
         g.setFont(new Font("default", Font.BOLD, 12));
+        long countOnSameSubway = entities.stream().filter(e -> getSubway(e) == playerSubway).count();
         for (Entity entity : entities) {
+            boolean isCat = entity.getType() == Type.CAT;
+            var image = Assets.getInstance().getCat();
+            int imageDown = isCat ? 0 : tileSize / 5;
+
+            // update the cat position if you see it:
+
+
+            // draw all entities on the same Subway or on the surface as this player
             if (getSubway(entity) == playerSubway || superVision) {
+                if (isCat) {
+                    catLastSeen.put(entity.getId(), entity.getPosition());
+                }
                 var bounds = g.getFontMetrics().getStringBounds(entity.getName(), g);
 
-                boolean isCat = entity.getId() == 0;
-                int imageDown = isCat ? 0 : tileSize / 5;
                 int textUp = tileSize / 3;
 
-                if (entity.isDead()) {
-                    var image = Assets.getInstance().getTombstone();
-                    textUp = 0;
-                    g.drawImage(image, tileSize * entity.getPosition().x(), tileSize * entity.getPosition().y(), tileSize, tileSize, null);
-                } else {
-                    var image = isCat ? Assets.getInstance().getCat() : Assets.getInstance().getMouse();
-                    g.drawImage(image, tileSize * entity.getPosition().x(), tileSize * entity.getPosition().y() + imageDown, tileSize, tileSize, null);
+                if (!isCat) {
+                    image = Assets.getInstance().getMouse();
+                    if (entity.isDead()) image = Assets.getInstance().getTombstone();
                 }
+                g.drawImage(image, tileSize * entity.getPosition().x(), tileSize * entity.getPosition().y() + imageDown, tileSize, tileSize, null);
```

Comment:
This change doesn't draw the tombstone in the same position anymore (it is a bit too far down)